modules/fleet-app-operator-permissions/variables.tf

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "project_id" {
+  description = "The project to which the Fleet belongs."
+  type        = string
+}
+
+variable "scope_id" {
+  description = "The scope for which IAM and RBAC role bindings are created."
+  type        = string
+}
+
+variable "app_operator_name" {
+  description = "The name of the app operator principal for the Fleet Scope, e.g., `person@google.com` (user), `people@google.com` (group), `principal://iam.googleapis.com/locations/global/workforcePools/my-pool/subject/person` (user), `principalSet://iam.googleapis.com/locations/global/workforcePools/my-pool/group/people` (group), `serviceAccount:my-service-account@my-project.iam.gserviceaccount.com` (user)."
+  type        = string
+}
+
+variable "is_user_app_operator" {
```

Comment:
Rather than introducing an additional variable, I'd suggest we try to simplify the user experience.  Looks like we required `user:person@google.com` or `group:people@google.com` for `app_operator_name`, this variable `is_user_app_operator` could be eliminated?

Could even add a variable validation to ensure the string starts with a valid prefix: `"user:", "group:", "principal:", "principalSet:"`

Reply:
I'm now using the `users` and `groups` variables, by which the lists of user and group app operators can be provided to have the specified role.

One thing about the format is that  a user/group is in fact an RBAC subject. The corresponding IAM principal is created accordingly, e.g., for user `person@google.com`, the IAM principal would be `user:person@google.com`. I defined these variables in RBAC style (rather than IAM style) to be consistent with how the same thing is done with [gcloud](https://cloud.google.com/sdk/gcloud/reference/alpha/container/fleet/scopes/add-app-operator-binding).  So as for validation, we don't have a set of acceptable prefixes with this approach.